src/client/app/components/maps/EditMapModalComponent.tsx

```diff
@@ -0,0 +1,151 @@
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import { useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Form, FormGroup, Label, Input } from 'reactstrap';
+import { Link } from 'react-router-dom';
+import { CalibrationModeTypes, MapMetadata } from '../../types/redux/map';
+import { showErrorNotification } from '../../utils/notifications';
+
+interface EditMapModalProps {
+	show: boolean;
+	handleClose: () => void;
+	map: MapMetadata;
+	editMapDetails(map: MapMetadata): any;
+	setCalibration(mode: CalibrationModeTypes, mapID: number): any;
+	removeMap(id: number): any;
+}
+
+/**
+ *Defines the edit maps modal form
+ * @param props state variables needed to define the component
+ * @returns Map edit element
+ */
+function EditMapModalComponent(props: EditMapModalProps) {
+	const [nameInput, setNameInput] = useState(props.map.name);
+	const [noteInput, setNoteInput] = useState(props.map.note || '');
+	const [circleInput, setCircleInput] = useState(props.map.circleSize.toString());
+	const [displayable, setDisplayable] = useState(props.map.displayable);
+
+	const intl = useIntl();
+
+	const handleSave = () => {
+		const updatedMap = {
+			...props.map,
+			name: nameInput,
+			note: noteInput,
+			circleSize: parseFloat(circleInput),
+			displayable: displayable
+		};
+		props.editMapDetails(updatedMap);
+		props.handleClose();
+	};
+
+	const handleDelete = () => {
+		const consent = window.confirm(intl.formatMessage({ id: 'map.confirm.remove' }, { name: props.map.name }));
+		if (consent) {
+			props.removeMap(props.map.id);
+			props.handleClose();
+		}
+	};
+
+	const handleCalibrationSetting = (mode: CalibrationModeTypes) => {
+		props.setCalibration(mode, props.map.id);
+		props.handleClose();
+	};
+
+	const toggleCircleEdit = () => {
+		const regtest = /^\d+(\.\d+)?$/;
+		if (regtest.test(circleInput) && parseFloat(circleInput) <= 2.0) {
+			setCircleInput(circleInput);
+		} else {
+			showErrorNotification(intl.formatMessage({ id: 'invalid.number' }));
+		}
+	};
+
+	return (
+		<Modal isOpen={props.show} toggle={props.handleClose}>
+			<ModalHeader toggle={props.handleClose}>
+				<FormattedMessage id="edit.map" />
+			</ModalHeader>
+			<ModalBody>
+				<Form>
+					<FormGroup>
+						<Label for="mapName"><FormattedMessage id="map.name" /></Label>
+						<Input
+							id="mapName"
+							value={nameInput}
+							onChange={e => setNameInput(e.target.value)}
+						/>
+					</FormGroup>
+					<FormGroup>
+						<Label for="mapDisplayable"><FormattedMessage id="map.displayable" /></Label>
+						<Input
+							id="mapDisplayable"
+							type="select"
+							value={displayable.toString()}
+							onChange={e => setDisplayable(e.target.value === 'true')}
+						>
+							<option value="true">{intl.formatMessage({ id: 'map.is.displayable' })}</option>
```

Comment:
This is done via an enum on other pages such as unit displayable. Also, I think this should mirror the bidirectional on conversion so yes/no with automatic coloring.